systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeIsolatedST.java

```diff
@@ -55,117 +54,121 @@ public class OlmUpgradeIsolatedST extends AbstractUpgradeST {
+        KafkaClients kafkaBasicClientJob = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+            .withTopicName(topicUpgradeName)
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(1000)
+            .build();
 
-        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi());
-        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi(), kafkaBasicClientJob.consumerStrimzi());
+        ClientUtils.waitForClientsSuccess(testStorage.getProducerName(), testStorage.getConsumerName(), clusterOperator.getDeploymentNamespace(), testStorage.getMessageCount());
 
         String clusterOperatorDeploymentName = ResourceManager.kubeClient().getDeploymentNameByPrefix(Environment.OLM_OPERATOR_DEPLOYMENT_NAME);
         LOGGER.info("Old deployment name of Cluster Operator is {}", clusterOperatorDeploymentName);
 
         // ======== Cluster Operator upgrade starts ========
         makeSnapshots();
-        // wait until non-used install plan is present (sometimes install-plan did not append immediately and we need to wait for at least 10m)
 
         OlmConfiguration upgradeOlmConfig = new OlmConfigurationBuilder(clusterOperator.getOlmResource().getOlmConfiguration())
             .withChannelName("stable")
             .withOperatorVersion(toVersion)
             .build();
 
         clusterOperator.getOlmResource().updateSubscription(upgradeOlmConfig);
-        // wait until non-used install plan is present (sometimes install-plan did not append immediately and we need to wait for at least 10m)
-        OlmUtils.waitUntilNonUsedInstallPlanIsPresent(INFRA_NAMESPACE, toVersion);
 
-        // Cluster Operator
-        clusterOperator.getOlmResource().upgradeClusterOperator();
-
-        // wait until RU is finished
-        zkPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), zkSelector, 3, zkPods);
-        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), kafkaSelector, 3, kafkaPods);
-        eoPods = DeploymentUtils.waitTillDepHasRolled(clusterOperator.getDeploymentNamespace(), KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
-        // ======== Cluster Operator upgrade ends ========
+        // Cluster Operator upgrade
+        clusterOperator.getOlmResource().upgradeClusterOperator(clusterOperator.getDeploymentNamespace(), upgradeOlmConfig.getCsvName());
```

Comment:
I would maybe add a possibility to update OLM configuration inside the `clusterOperator` and remove the parameters from `upgradeClusterOperator()` and `updateSubscription()`. But it's just an idea.